src/axolotl/core/trainer_builder.py

```diff
@@ -735,7 +735,7 @@ def build(self, total_num_steps):
         elif self.cfg.sample_packing and self.cfg.eval_sample_packing is False:
             training_arguments_kwargs["dataloader_drop_last"] = True
 
-        if self.cfg.val_set_size == 0:
+        if not hasattr(self.cfg, "test_datasets") and self.cfg.val_set_size == 0:
```

Comment:
I believe it is probably enough to write:

```
if not cfg.test_datasets and self.cfg.val_set_size == 0:
```

Reply:
```suggestion
        if not cfg.test_datasets and self.cfg.val_set_size == 0:
```